src/status_im/contexts/communities/actions/addresses_for_permissions/view.cljs

```diff
@@ -263,7 +237,106 @@
                                               (rf/dispatch [:hide-bottom-sheet]))}]))}])
             (do
               (rf/dispatch [:communities/set-share-all-addresses id flag-share-all-addresses])
-              (rf/dispatch [:communities/set-addresses-to-reveal id addresses-to-reveal]))))]
+              (rf/dispatch [:communities/set-addresses-to-reveal id addresses-to-reveal]))))
+        pending? (rf/sub [:communities/has-pending-request-to-join? id])
+        highest-role (rf/sub [:communities/highest-role-for-selection id])
+        [unmodified-role _] (rn/use-state highest-role)]
+    [quo/bottom-actions
+     (cond-> {:role             (role-keyword highest-role)
+              :description      (if highest-role
+                                  :top
+                                  :top-error)
+              :button-one-props {:customization-color color
+                                 :on-press            confirm-changes
+                                 :disabled?           (or checking?
+                                                          (empty? addresses-to-reveal)
+                                                          (not highest-role)
+                                                          identical-choices?)}}
```

Comment:
Is this the flickering button?
If you want I can create a separate issue for resolving this glitch. I'm wondering if refactoring away from `use-effect` would resolve this or if the problem will persist because of the two state changes not being batched. 🤔

Reply:
let me check, I can probably give it a quick look, and then if it's merged like this, we can create a separate issue.